marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java

```diff
@@ -254,18 +228,17 @@ private void initializeFlow(String jobId) {
     }
 
     public void stopJob(String jobId) {
-        synchronized (stepsMap) {
-            if(stepsMap.get(jobId) != null){
-                stepsMap.get(jobId).clear();
-                stepsMap.remove(jobId);
-                isJobCancelled.set(true);
-            }
-            else {
-                throw new RuntimeException("Job not running");
-            }
+        if (stepsMap.get(jobId) != null) {
```

Comment:
What was the issue with "synchronized"? I don't know if it's actually needed, but I also don't know the impact of removing it.

Reply:
Coverity had raised a concurrency issue earlier which is why @ryanjdew added this in. Right now it's raising an issue that we are synchronizing over an object of java.util.concurrent package.

I checked and also conformed with Ryan that its safe to remove the synchronized block since both stepsMap and isJobCancelled are part of java.util.concurrent package.

Ryan and I decided to remove the synchronized block for now and see if the Coverity scan raises the earlier issue again.